src/dbnode/encoding/istream.go

```diff
@@ -29,9 +29,8 @@ import (
 // istream encapsulates a readable stream.
 type istream struct {
 	r         *bufio.Reader // encoded stream
-	err       error         // error encountered
```

Comment:
Why are we removing the error? Shouldn't ReadBit still fail if the stream has failed?

Reply:
+1

Reply:
This error is set from the result of `readByteFromStream`, but in all of the cases we call this, the parent call also returns that error (e.g. `ReadBit`). So I see no reason to keep the error stored as state on the stream object itself if we already would have returned it. Is there a reason to keep it, though, that I'm missing? The value of removing it is that we can then remove these if-conditions that are in multiple methods to check if the error is set.

Reply:
Is there a noticeable difference on flame graphs? Branch prediction should help reduce the cost of these when there's no error

Reply:
Yeah it is small but noticeable on the flamegraph. Most of `ReadBits` is attributable to `readByte`, and that increases when we remove this err checks (meaning less CPU work in the `ReadBits` func).

BEFORE
![image](https://user-images.githubusercontent.com/9585325/76254734-5bb22380-6223-11ea-82a5-1ed44a5a6d7f.png)

AFTER
![image](https://user-images.githubusercontent.com/9585325/76254793-71bfe400-6223-11ea-984b-cb15db41069c.png)

Reply:
I think I'd still lean on the side of caution here; otherwise there's a bit of a weird reliance that `is.r.Byte()` will error in the expected fashion otherwise, and if the underlying reader changes you may get weird behaviour

Reply:
Ok sounds good to me. I'll update to only include the casting changes.